### DISCUSSION TITLE
How to build release like binary

### DISCUSSION THREAD

**Discussion by lhoff94** [micropython/micropython#9198] [Using MicroPython] [answered]
Hi, 

I build the esp32 build for the first time today and tried flashing it like I'm used to it with esptool but it didn't work.
`esptool.py --chip esp32 --port /dev/ttyUSB0 write_flash -z 0x1000 micropython.bin`
The same command works with the binaries from the official website. 
Using the commandline that idf gave after the build was complete (with bootloader and partition table) worked. 
I assume the binaries from the website come with the bootloader and partition table included. How can I do this myself?

I tried finding a github action within the micropython repo that triggers the release build and contains the command but couldn't find something. 

Thanks in advance

**Comment by jimmo** [comment, marked as answer]
> ... -z 0x1000 micropython.bin

You need to use `firmware.bin` which is the combined firmware that you're talking about. See instructions at

https://github.com/micropython/micropython/blob/master/ports/esp32/README.md#building-the-firmware

**Comment by lhoff94** [comment]
Oh Well. I shouldn't ask questions when I'm tired in the middle of the night. 
I read the README multiple times but missed or misunderstood that part. 
Thanks.
